packages/eslint-plugin/lib/rules/require-img-alt.js

```diff
@@ -60,14 +66,73 @@ module.exports = {
         if (node.name !== "img") {
           return;
         }
-        if (!hasAltAttrAndValue(node, substitute)) {
-          context.report({
-            loc: {
-              start: node.openStart.loc.start,
-              end: node.openEnd.loc.end,
-            },
-            messageId: MESSAGE_IDS.MISSING_ALT,
-          });
+
+        const altResult = hasValidAltOrSubstitute(node, substitute);
+        const hasSubstituteOption = substitute.length > 0;
+
+        if (!altResult.hasAnyAlt) {
+          if (hasSubstituteOption) {
+            context.report({
+              loc: {
+                start: node.openStart.loc.start,
+                end: node.openEnd.loc.end,
+              },
+              messageId: MESSAGE_IDS.MISSING_ALT,
+            });
+          } else {
+            context.report({
+              loc: {
+                start: node.openStart.loc.start,
+                end: node.openEnd.loc.end,
+              },
+              messageId: MESSAGE_IDS.MISSING_ALT,
+              suggest: [
+                {
+                  messageId: MESSAGE_IDS.INSERT_ALT,
+                  fix(fixer) {
+                    return fixer.insertTextBefore(node.openEnd, ' alt=""');
+                  },
+                },
+              ],
+            });
+          }
+        } else if (altResult.hasEmptyAlt && !altResult.hasValidContent) {
+          if (hasSubstituteOption) {
+            context.report({
+              loc: {
+                start: node.openStart.loc.start,
+                end: node.openEnd.loc.end,
+              },
+              messageId: MESSAGE_IDS.EMPTY_ALT,
+            });
+          } else {
+            context.report({
+              loc: {
+                start: node.openStart.loc.start,
+                end: node.openEnd.loc.end,
+              },
+              messageId: MESSAGE_IDS.EMPTY_ALT,
+              suggest: [
+                {
+                  messageId: MESSAGE_IDS.INSERT_ALT,
+                  fix(fixer) {
+                    const emptyAltAttr = node.attributes.find(
```

Comment:
The checks we did above are being done again in the fixer. This whole part seems unnecessary.